src/slurm_plugin/instance_manager.py

```diff
@@ -358,3 +361,90 @@ def terminate_all_compute_nodes(self, terminate_batch_size):
     def _update_failed_nodes(self, nodeset, error_code="Exception"):
         """Update failed nodes dict with error code as key and nodeset value."""
         self.failed_nodes[error_code] = self.failed_nodes.get(error_code, set()).union(nodeset)
+
+    def get_console_output_task(self, nodes, output_callback, wait_times):
+        """
+        Return a function that will retrieve the console output for each node in `nodes`.
+
+        The console output for each node will be passed to output_callback.
+        """
+        start_time = datetime.now(tz=timezone.utc)
+
+        region = self._region
+        instances = tuple(self._get_instances_for_nodes(nodes))
+        if len(instances) < 1:
+            return None
+
+        def report_console_output(ec2client, output_hashes):
+            for output in InstanceManager._get_console_output_from_nodes(ec2client, instances):
+                instance_id, console_output = output.get("InstanceId"), output.get("ConsoleOutput")
+                current_hash = hashlib.sha256(console_output.encode()).hexdigest() if console_output else None
+                previous_hash = output_hashes.get(instance_id)
+                if current_hash != previous_hash:
+                    output_callback(output)
+                    output_hashes.update({instance_id: current_hash})
+
+        def console_collector():
+            try:
+                ec2 = boto3.session.Session().client("ec2", region_name=region)
+
+                output_hashes = {}
+
+                for next_wait in wait_times:
```

Comment:
I was thinking we might want to grab the log at the time we detect the node is down and then collect it again after the wait time in an attempt to get extra info in case the final output went over the 64K mark.

Removed.